Add Golang at Choose Driver or ORM (#10529)
pingcap/docs: develop/dev-guide-choose-driver-or-orm.md

## Java

> This section describes how to use Driver and ORM in Java.

Reviewer: ```suggestion
This section describes how to use drivers and ORM frameworks in Java.
```

> This section describes how to use Driver and ORM of Java language.

Reviewer 2: ```suggestion
This section describes how to use Driver and ORM in Java.
```

> ### Java Drivers

Reviewer: ```suggestion
### Java drivers
```

> ### Java ORM framework

Reviewer: ```suggestion
### Java ORM frameworks
```

> **Note:**
>
> - Currently, Hibernate does [not support nested transactions](https://stackoverflow.com/questions/37927208/nested-transaction-in-spring-app-with-jpa-postgres).

> > - Since v6.2.0, TiDB supports [savepoint](/sql-statements/sql-statement-savepoint.md). Therefore, make sure that your TiDB is v6.2.0 or later versions. Otherwise, do not use the `Propagation.NESTED` transaction propagation option in `@Transactional`, that is, do not set `@Transactional(propagation = Propagation.NESTED)`.

Reviewer: ```suggestion
> - Since v6.2.0, TiDB supports [savepoint](/sql-statements/sql-statement-savepoint.md). To use the `Propagation.NESTED` transaction propagation option in `@Transactional`, that is, to set `@Transactional(propagation = Propagation.NESTED)`, make sure that your TiDB is v6.2.0 or later.
```

>
> If you are unable to upgrade your `Hibernate` version, use the MySQL 5.7 dialect `org.hibernate.dialect.MySQL57Dialect` instead. However, this setting might cause unpredictable results and the absence of some TiDB-specific features, such as [sequences](/sql-statements/sql-statement-create-sequence.md).

</div>

<div label="MyBatis">

Support level: **Full**

> To avoid manually managing complex relationships between different dependencies of an application, you can use [Gradle](https://gradle.org/install) or [Maven](https://maven.apache.org/install.html) to get all dependencies of your application, including those indirect ones.

Reviewer 2: ```suggestion
To avoid manually managing complex relationships between different dependencies of an application, you can use [Gradle](https://gradle.org/install) or [Maven](https://maven.apache.org/install.html) to get all dependencies of your application, including those indirect dependencies.
```

If you are using Maven, add the following to your `<dependencies></dependencies>`:

```xml
<dependency>
    <groupId>org.mybatis</groupId>
    <artifactId>mybatis</artifactId>
    <version>3.5.9</version>
</dependency>

<dependency>
    <groupId>mysql</groupId>
    <artifactId>mysql-connector-java</artifactId>
    <version>5.1.49</version>
</dependency>
```

If you are using Gradle, add the following to your `dependencies`:

```gradle
implementation 'org.mybatis:mybatis:3.5.9'
implementation 'mysql:mysql-connector-java:5.1.49'
```

For an example of using MyBatis to build a TiDB application, see [Build a Simple CRUD App with TiDB and Java](/develop/dev-guide-sample-application-java.md).

</div>

</SimpleTab>

> ### Java Client Load Balancing

Reviewer: ```suggestion
### Java client load balancing
```

implementation group: 'io.github.lastincisor', name: 'mysql-connector-java', version: '8.0.29-tidb-1.0.0'
implementation group: 'io.github.lastincisor', name: 'tidb-loadbalance', version: '0.0.5'
```

## Golang

> This section describes how to use Driver and ORM in Golang.

Reviewer: ```suggestion
This section describes how to use drivers and ORM frameworks in Golang.
```

> This section describes how to use Driver and ORM of Golang language.

Reviewer 2: ```suggestion
This section describes how to use Driver and ORM in Golang.
```

> ### Golang Drivers

Reviewer: ```suggestion
### Golang drivers
```

**go-sql-driver/mysql**

Support level: **Full**

> You can follow the [go-sql-driver/mysql documentation](https://github.com/go-sql-driver/mysql) to download and configure a Golang driver.

Reviewer 2: ```suggestion
To download and configure a Golang driver, refer to the [go-sql-driver/mysql documentation](https://github.com/go-sql-driver/mysql).
```

For an example of how to build a complete application, see [Build a Simple CRUD App with TiDB and Golang](/develop/dev-guide-sample-application-golang.md).

> ### Golang ORM Framework

Reviewer: ```suggestion
### Golang ORM frameworks
```

**GORM**

Support level: **Full**

> GORM is a popular ORM framework for Golang, and you can use `go get` to get all of your application's dependencies.

Reviewer 2: ```suggestion
GORM is a popular ORM framework for Golang. To get all dependencies in your application, you can use the `go get` command.
```
